lab_2_tokenize_by_bpe/main.py

```diff
@@ -14,10 +14,22 @@ def prepare_word(
     :param end_of_word: a token that signifies the end of word
     :return: preprocessed word
     """
+    if not isinstance(raw_word, str) or (start_of_word is not None and
+                                         not isinstance(start_of_word, str)) or (
+            end_of_word is not None and not isinstance(end_of_word, str)):
+        return None
+    total_result = []
+    if start_of_word:
```

Comment:
Theoretically, `start_of_word` and `end_of_word` could be empty stings. In this case, such condition will not produce the desired output.

Reply:
I'll work on it

Reply:
As `None` is a singleton, correct operator here would be `is`